nixops/resources/aws_vpn_connection_route.py

```diff
@@ -0,0 +1,126 @@
+# -*- coding: utf-8 -*-
+
+import botocore
+
+import nixops.util
+import nixops.resources
+from nixops.resources.ec2_common import EC2CommonState
+import nixops.ec2_utils
+from nixops.diff import Diff, Handler
+from nixops.state import StateDict
+
+class AWSVPNConnectionRouteDefinition(nixops.resources.ResourceDefinition):
+    """Definition of a VPN connection route"""
+
+    @classmethod
+    def get_type(cls):
+        return "aws-vpn-connection-route"
+
+    @classmethod
+    def get_resource_type(cls):
+        return "awsVPNConnectionRoutes"
+
+    def show_type(self):
+        return "{0}".format(self.get_type())
+
+class AWSVPNConnectionState(nixops.resources.ResourceState, EC2CommonState):
+    """State of a VPN connection route"""
+
+    state = nixops.util.attr_property("state", nixops.resources.ResourceState.MISSING, int)
+    access_key_id = nixops.util.attr_property("accessKeyId", None)
+    _reserved_keys = EC2CommonState.COMMON_EC2_RESERVED
+
+    @classmethod
+    def get_type(cls):
+        return "aws-vpn-connection-route"
+
+    def __init__(self, depl, name, id):
+        nixops.resources.ResourceState.__init__(self, depl, name, id)
+        self._client = None
+        self._state = StateDict(depl, id)
+        self.region = self._state.get('region', None)
+        self.handle_create_vpn_route = Handler(['region', 'vpnConnectionId', 'destinationCidrBlock'], handle=self.realize_create_vpn_route)
+
+    def show_type(self):
+        s = super(AWSVPNConnectionState, self).show_type()
+        if self.region: s = "{0} [{1}]".format(s, self.region)
+        return s
+
+    @property
+    def resource_id(self):
+        return self._state.get('destinationCidrBlock', None)
+
+    def prefix_definition(self, attr):
+        return {('resources', 'awsVPNConnectionRoutes'): attr}
+
+    def get_definition_prefix(self):
+        return "resources.awsVPNConnectionRoutes."
```

Comment:
Can't `get_definition_prefix` be derived from `prefix_definition`, or vice versa?

Reply:
In fact, given that `AWSVPNConnectionRouteDefinition.get_resource_type` already returns `awsVPNConnectionRoutes`, maybe these methods can be avoided entirely?